library/private/options_tools.h

```diff
@@ -191,6 +194,46 @@ ratio_t parse(const std::string& str)
 template<>
 color_t parse(const std::string& str)
 {
+  std::string s = options_tools::trim(str);
+  std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) { return std::tolower(c); });
+
+  /* Hex format search */
+  const std::regex hexRegex("^#([0-9a-fA-F]{6})$");
+  std::smatch hexMatch;
+  if (std::regex_search(s, hexMatch, hexRegex))
+  {
+    std::vector<double> vec;
+    const std::string hexStr = hexMatch[1];
+    for (size_t i = 0; i < hexStr.length(); i += 2)
+    {
+      unsigned int byte = std::stoul(hexStr.substr(i, 2), nullptr, 16);
+      vec.push_back(byte / 255.0);
+    }
+    return color_t(vec);
+  }
```

Comment:
```suggestion
  const std::regex hexRegex(
    "^#([0-9a-f]{2})([0-9a-f]{2})([0-9a-f]{2})$", std::regex_constants::icase);
  std::smatch hexMatch;
  if (std::regex_search(s, hexMatch, hexRegex))
  {
    return color_t(                                 //
      std::stoul(hexMatch[1], nullptr, 16) / 255.0, //
      std::stoul(hexMatch[2], nullptr, 16) / 255.0, //
      std::stoul(hexMatch[3], nullptr, 16) / 255.0  //
    );
  }
```
using a group for each component is maybe more straightforward? also using `icase` can avoid having to specify `a-fA-F`